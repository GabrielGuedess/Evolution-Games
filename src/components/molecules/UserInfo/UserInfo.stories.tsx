import * as Popover from '@radix-ui/react-popover';
import { Story, Meta } from '@storybook/react/types-6-0';

import { userInfo } from 'templates/Home/mock';

import { AvatarUserProps } from 'components/molecules/AvatarUser/AvatarUser';

import { UserInfo } from './UserInfo';

export default {
  title: 'Molecules/UserInfo',
  component: UserInfo,
  args: userInfo,
  argTypes: {
    name: {
      type: 'string',
    },
    bio: {
      type: 'string',
    },
    username: {
      type: 'string',
    },
    userPhoto: {
      type: 'string',
    },
  },
} as Meta<AvatarUserProps>;

export const Default: Story = args => (
  <Popover.Root>
    <UserInfo {...args} />
  </Popover.Root>
);
