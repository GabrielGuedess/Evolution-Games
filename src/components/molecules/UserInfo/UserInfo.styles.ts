import * as Avatar from '@radix-ui/react-avatar';

import styled, { css, keyframes } from 'styled-components';

const rotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    min-width: 40rem;
    width: 40rem;
    padding: ${theme.spacings.large};
    background: ${theme.colors.gameDetails};
    border-radius: ${theme.border.radius.regular};
    display: flex;
    flex-direction: column;
    gap: 2.4rem;
    position: relative;
  `}
`;

export const Root = styled(Avatar.Root)`
  display: flex;
  justify-content: center;
`;

export const WrapperPhoto = styled.div`
  ${({ theme }) => css`
    position: relative;

    ::before,
    ::after {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      border-radius: 50%;
      border-top: 0.2rem solid ${theme.colors.white};
      z-index: ${theme.layers.base};
      filter: drop-shadow(0 0 2px ${theme.colors.primary})
        drop-shadow(0 0 5px ${theme.colors.primary})
        drop-shadow(0 0 10px ${theme.colors.primary})
        drop-shadow(0 0 20px ${theme.colors.primary});
      animation: ${rotate} 5s infinite linear;
    }

    ::after {
      filter: drop-shadow(0 0 2px #f554ff) drop-shadow(0 0 5px #f554ff)
        drop-shadow(0 0 10px #f554ff) drop-shadow(0 0 20px #f554ff);
      animation-delay: -2.5s;
    }
  `}
`;
export const AvatarImage = styled(Avatar.Image)`
  width: 12rem;
  aspect-ratio: 1/1;
  border-radius: 50%;
  object-fit: cover;
`;

export const UserInfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1.6rem;
`;

export const Name = styled.strong`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xbig};
    color: ${theme.colors.whiteText};
    font-weight: 500;
    text-align: center;
  `}
`;

export const Username = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.secondary};

    span {
      color: ${theme.colors.primary};
    }
  `}
`;

export const Bio = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.whiteText};
    line-height: 2rem;
  `}
`;

export const ButtonWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;
