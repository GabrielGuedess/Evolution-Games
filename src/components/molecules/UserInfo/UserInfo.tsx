import Link from 'next/link';

import * as Avatar from '@radix-ui/react-avatar';

import Button from 'components/atoms/Button/Button';
import { CloseButton } from 'components/atoms/CloseButton/CloseButton';
import { AvatarUserProps } from 'components/molecules/AvatarUser/AvatarUser';

import * as S from './UserInfo.styles';

export const UserInfo = ({
  name,
  bio,
  username,
  userPhoto,
}: AvatarUserProps) => (
  <S.Wrapper>
    <CloseButton />

    <S.Root>
      <S.AvatarImage src={userPhoto!} alt="Image User" />
      <Avatar.Fallback />
    </S.Root>

    <S.UserInfoWrapper>
      <S.Name>{name}</S.Name>
      <S.Username>
        <span>@</span>
        {username}
      </S.Username>
      <S.Bio>{bio}</S.Bio>

      <S.ButtonWrapper>
        <Link href="/profile/me" passHref>
          <Button as="a">View Profile</Button>
        </Link>
      </S.ButtonWrapper>
    </S.UserInfoWrapper>
  </S.Wrapper>
);
