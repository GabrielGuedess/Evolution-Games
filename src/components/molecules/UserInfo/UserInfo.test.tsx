import * as Popover from '@radix-ui/react-popover';
import { renderWithTheme } from 'utils/tests/helpers';

import { UserInfo } from './UserInfo';

describe('<UserInfo />', () => {
  it('should render the heading', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Popover.Root>
        <UserInfo />
      </Popover.Root>,
    );

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
