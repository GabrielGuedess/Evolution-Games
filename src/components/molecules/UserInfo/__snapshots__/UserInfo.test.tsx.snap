// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<UserInfo /> should render the heading 1'] = `
.c8 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  position: relative;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 5px;
  -webkit-transition: all 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: all 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  background: #855aff;
  color: #F4F4F5;
}

.c8:not(:disabled):is(:hover,:focus) {
  box-shadow: 0 0.8rem 2.8rem rgba(118,74,241,0.5);
  color: #F4F4F5;
}

.c8 .effect {
  fill: #855aff;
}

.c8:not(:disabled):is(:hover,:focus) {
  -webkit-filter: brightness(1.1);
  filter: brightness(1.1);
}

.c8:not(:disabled):active {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c8 span {
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 1;
  display: block;
  position: relative;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  -webkit-transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  z-index: 1;
}

.c8:not(:disabled):is(:hover,:focus) span {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c8:not(:disabled):is(:hover,:focus) .icon {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c8:not(:disabled):active span {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c8 svg {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  -webkit-transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
}

.c8 .effect {
  position: absolute;
  top: -5%;
  left: -5%;
  width: 110%;
  height: 110%;
  z-index: -1;
}

.c8 .effect path {
  -webkit-transition: 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
}

.c8:not(:disabled):is(:hover,:focus) .effect path {
  d: path( 'M0,0 C0,-5 100,-5 100,0 C105,0 105,100 100,100 C100,105 0,105 0,100 C-5,100 -5,0 0,0' );
}

.c8:not(:disabled):active .effect path {
  d: path( 'M0 5C0 2.23858 2.23858 0 5 0H95C97.7614 0 100 2.23858 100 5V95C100 97.7614 97.7614 100 95 100H5C2.23858 100 0 97.7614 0 95V5Z' );
}

.c1 {
  color: #855aff;
  padding: 0.5rem;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  font-size: 1.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  z-index: 50;
  top: 0.5rem;
  left: 2.1rem;
}

.c1 svg {
  cursor: pointer;
}

.c0 {
  min-width: 40rem;
  width: 40rem;
  padding: 4.0rem;
  background: #010914;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.4rem;
  position: relative;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
}

.c4 {
  font-size: 2.5rem;
  color: #F4F4F5;
  font-weight: 500;
  text-align: center;
}

.c5 {
  font-size: 1.2rem;
  color: #717B8A;
}

.c5 span {
  color: #855aff;
}

.c6 {
  font-size: 1.4rem;
  color: #F4F4F5;
  line-height: 2rem;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c1 {
    left: 0.5rem;
  }
}

<div
  class="c0"
>
  <button
    aria-label="Close Action"
    class="c1"
    color="primary"
    side="left"
    type="button"
  >
    <svg
      fill="currentColor"
      height="20"
      viewBox="0 0 256 256"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <line
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="200"
        x2="56"
        y1="56"
        y2="200"
      />
      <line
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="200"
        x2="56"
        y1="200"
        y2="56"
      />
    </svg>
  </button>
  <span
    class="c2"
  >
    <span />
  </span>
  <div
    class="c3"
  >
    <strong
      class="c4"
    />
    <span
      class="c5"
    >
      <span>
        @
      </span>
    </span>
    <span
      class="c6"
    />
    <div
      class="c7"
    >
      <a
        aria-label="Button Action"
        class="c8"
        href="/profile/me"
      >
        <span>
          View Profile
        </span>
        <svg
          aria-hidden="true"
          class="effect"
          preserveAspectRatio="none"
          viewBox="-5 -5 110 110"
        >
          <path
            d="M0 8C0 3.58172 3.58172 0 8 0H92C96.4183 0 100 3.58172 100 8V92C100 96.4183 96.4183 100 92 100H8C3.58172 100 0 96.4183 0 92V8Z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
